waiter/integration/waiter/metrics_output_test.clj

```diff
@@ -140,3 +140,71 @@
+     (is (number? p-value#) (str "missing p" p# " value"))
+     p-value#))
+
+(defn- one-started-instance-observed?
+  "Returns true if the launch-metrics state reflects a single started instance
+   in the instance counts for the given service on the given router."
+  [router-url service-id]
+  (-> (make-request router-url "/state/launch-metrics")
+      :body
+      json/read-str
+      (get-in ["state" "service-id->launch-tracker" service-id "instance-counts" "started"])
+      (= 1)))
+
+(deftest ^:parallel ^:integration-slow test-launch-metrics-output
+  (testing-using-waiter-url
+    (let [waiter-settings (waiter-settings waiter-url)
+          metrics-sync-interval-ms (get-in waiter-settings [:metrics-config :metrics-sync-interval-ms])
+          router->endpoint (routers waiter-url)
+          router-urls (vals router->endpoint)
+          service-name (rand-name)
+          sleep-seconds 20
+          min-startup-seconds 10 ; 20s +/- 10s for 2 polls with 5s granularity
+          max-startup-seconds 60 ; the service shouldn't take more than a minute to become healthy
+          instance-count 2
+          req-headers {:x-waiter-cmd (str "sleep " sleep-seconds " && " (kitchen-cmd "-p $PORT0"))
+                       :x-waiter-cmd-type "shell"
+                       :x-waiter-min-instances instance-count
+                       :x-waiter-name service-name}
+          {:keys [headers request-headers service-id] :as first-response}
+          (make-request-with-debug-info req-headers #(make-kitchen-request waiter-url % :method :get))]
+      (with-service-cleanup
+        service-id
+        ; ensure the first request succeded before continuing with testing
+        (assert-response-status first-response 200)
+        ; check that the launch-metrics on each router are present and have sane values
+        (doseq [[router-id router-url] router->endpoint]
+          (wait-for #(one-started-instance-observed? router-url service-id)
+                    :interval 1 :timeout 10)
+          (let [metrics-response (->> "/metrics"
+                                      (make-request router-url)
+                                      :body
+                                      json/read-str)
+                service-launch-metric-timers (get-in metrics-response ["services" service-id "timers" "launch-overhead"])
+                service-scheduling-timer (get service-launch-metric-timers "schedule-time")
+                service-startup-timer (get service-launch-metric-timers "startup-time")
+                waiter-scheduling-timer (get-in metrics-response ["waiter" "launch-overhead" "timers" "schedule-time"])]
+            (testing "all launch metrics present"
+              (is (every? some? [service-scheduling-timer service-startup-timer waiter-scheduling-timer])))
+            (testing "expected launch-metric instance counts"
+              (is (== instance-count
+                      (get service-scheduling-timer "count")))
```

Comment:
Question: Why does waiting for one instance to become available mean we will have `instance-count` values available?

Reply:
Good catch. This is definitely racy as-is. Originally I had it hard-coded to wait for 10 seconds or something like that, and then changed it in the last round of feedback. I believe I was thinking there was only once instance total (which was wrong, since `instance-count` is `2`). It now waits for `instance-count` started instances.